src/components/NewCommentForm.tsx

```diff
@@ -47,25 +66,30 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
   const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    setErrors({
+    const newErrors = {
       name: !name,
       email: !email,
       body: !body,
-    });
+      submit: false,
+    };
 
-    if (!name || !email || !body) {
+    setErrors(newErrors);
+
+    if (newErrors.name || newErrors.email || newErrors.body) {
       return;
     }
 
     setSubmitting(true);
-
-    // it is very easy to forget about `await` keyword
-    await onSubmit({ name, email, body });
-
-    // and the spinner will disappear immediately
-    setSubmitting(false);
-    setValues(current => ({ ...current, body: '' }));
-    // We keep the entered name and email
+    setErrors(current => ({ ...current, submit: false }));
+
+    try {
+      await createNewComment({ name, email, body });
```

Comment:
It's good that you are using async/await with try/catch for error handling. However, you don't seem to be doing anything with the error object in your catch block. It would be a good idea to at least log the error, or better yet, show a more specific error message to the user based on the type of error.